include/dl/dl.h

```diff
@@ -411,6 +411,70 @@ typedef struct dl_instance_info
 */
 dl_error_t DL_DLL_EXPORT dl_instance_get_info( const unsigned char* packed_instance, size_t packed_instance_size, dl_instance_info_t* out_info );
 
+/*
+	Function: dl_get_metadata_count
+	    Fetch the number of meta data instances for the specified type, enum or union
+
+	Parameters:
+	    dl_ctx          - Context where type-info for type and metadata is stored.
+	    type            - Type id for type to fetch the meta data count of.
+	    count           - The number of meta data instances available for the give type
+
+	Return:
+	    DL_ERROR_OK on success.
+*/
+dl_error_t DL_DLL_EXPORT dl_get_metadata_count( dl_ctx_t dl_ctx, dl_typeid_t type, unsigned int* count );
+
+/*
+	Function: dl_get_member_metadata_count
+	    Fetch the number of meta data instances for the specified type member, enumerator or union member
+
+	Parameters:
+	    dl_ctx          - Context where type-info for type and metadata is stored.
+	    type            - Type id for type to fetch the meta data count of.
+	    item_index      - The index of the member, enumerator or union member
+	    count           - The number of meta data instances available for the give type
+
+	Return:
+	    DL_ERROR_OK on success.
+*/
+dl_error_t DL_DLL_EXPORT dl_get_member_metadata_count( dl_ctx_t dl_ctx, dl_typeid_t type, unsigned int item_index, unsigned int* count );
+
+/*
+	Function: dl_get_metadata
+		Fetch the number of meta data instances for the specified type
+
+	Parameters:
+		dl_ctx            - Context where type-info for type and metadata is stored.
+		type              - Type id for type to fetch the meta data instance of.
+		index             - The index of the meta data to fetch
+		out_type_id       - The type id of the returned meta data instance
+	    out_instance_size - The size of the returned instance
+		out_instance      - A pointer to the instance
+
+	Return:
+		DL_ERROR_OK on success.
+*/
+dl_error_t DL_DLL_EXPORT dl_get_metadata( dl_ctx_t dl_ctx, dl_typeid_t type, unsigned int index, dl_typeid_t* out_type_id, void** out_instance );
```

Comment:
Now it strike me, do we need 2 functions for these?
Probably, if you need one you need all of them. Would it make more sens to have one function to fill arrays?
I.e. passning a pointer to an array to fill, size of passet array and a pointer where to return the metadata count?

Not sure, but it might be worth thinking about before comitting to this api

Reply:
I'm starting to think that this type related data is really extended reflection information, and thus could be merged with `dl_type_info`, `dl_member_info`, `dl_enum_info` and `dl_enum_value_info`. So no extra getters, you get them with the dl_reflect functions which are already there.

Brilliant or brown? Perfect or poop?

Reply:
hmm... I think that sounds right yes. Less api to read, less api to understand... I like it!

Reply:
The move is done